clippy_lints/src/casts/borrow_as_ptr.rs

```diff
@@ -23,6 +25,10 @@ pub(super) fn check<'tcx>(
         };
         let mut app = Applicability::MachineApplicable;
         let snip = snippet_with_context(cx, e.span, cast_expr.span.ctxt(), "..", &mut app).0;
+        // Fix #9884
+        if for_each_unconsumed_temporary(cx, expr, |_| ControlFlow::Break(true)).is_break() {
```

Comment:
We can copy the criteria rustc uses - https://github.com/rust-lang/rust/blob/3ed2a10d173d6c2e0232776af338ca7d080b1cd4/compiler/rustc_hir_typeck/src/expr.rs#L482-L499

Reply:
I was looking for that for quite a while, only finding that utils function. Thank you!

Reply:
A good tip is to search for the error message, here "cannot take address of a temporary", and work your way up from that

Reply:
I often do that after initially searching the docs, so it's a bit weird I couldn't find it then